lib/polymorphic_integer_type/activerecord_5_0_0/polymorphic_array_value_extension.rb

```diff
@@ -1,17 +1,31 @@
 module PolymorphicIntegerType
   module PolymorphicArrayValueExtension
+
+    # original method:
+    # def type_to_ids_mapping
+    #   default_hash = Hash.new { |hsh, key| hsh[key] = [] }
+    #   result = values.each_with_object(default_hash) do |value, hash|
+    #     hash[klass(value).polymorphic_name] << convert_to_id(value)
+    #   end
+    # end
+
     def type_to_ids_mapping
-      super.tap do |result|
-        association = @associated_table.send(:association)
-        klass = association.active_record
-        name = association.name
+      association = @associated_table.send(:association)
+      name = association.name
+      default_hash = Hash.new { |hsh, key| hsh[key] = [] }
+      values.each_with_object(default_hash) do |value, hash|
+        klass = respond_to?(:klass) ? klass(value) : klass.class
+        if association.active_record.respond_to?("#{name}_type_mapping")
+          mapping = association.active_record.send("#{name}_type_mapping")
+          key ||= mapping.key(klass.polymorphic_name) if klass.respond_to?(:polymorphic_name)
```

Comment:
This is the meat of the change. We don't know whether users of the gem have stored their namespaces in the mappings or not, so we'll try several methods till we get a hit in the mapping.